WordPress/Classes/Services/BlogService.m

```diff
@@ -220,23 +220,37 @@ - (void)syncSettingsForBlog:(Blog *)blog
                     success:(void (^)())success
                     failure:(void (^)(NSError *error))failure
 {
-    id<BlogServiceRemote> remote = [self remoteForBlog:blog];
-    [remote syncSettingsForBlog:blog
-                        success:^(RemoteBlogSettings *settings) {
-                            blog.blogName = settings.name;
-                            blog.blogTagline = settings.desc;
-                            if (settings.defaultCategory) {
-                                blog.defaultCategoryID = settings.defaultCategory;
-                            }
-                            if (settings.defaultPostFormat) {
-                                blog.defaultPostFormat = settings.defaultPostFormat;
-                            }
-                            [self.managedObjectContext save:nil];
-                            if (success) {
-                                success();
-                            }
-                        }
-                        failure:failure];
+    NSManagedObjectID *blogID = [blog objectID];
+    [self.managedObjectContext performBlock:^{
```

Comment:
I see this being done in a lot of places. This is why I wanted to get a
standard on services...

On Thursday, 27 August 2015, Aaron Douglas <notifications@github.com> wrote:

> In WordPress/Classes/Services/BlogService.m
> <https://github.com/wordpress-mobile/WordPress-iOS/pull/4182#discussion_r38086245>
> :
>
> > -                            blog.blogName = settings.name;
> > -                            blog.blogTagline = settings.desc;
> > -                            if (settings.defaultCategory) {
> > -                                blog.defaultCategoryID = settings.defaultCategory;
> > -                            }
> > -                            if (settings.defaultPostFormat) {
> > -                                blog.defaultPostFormat = settings.defaultPostFormat;
> > -                            }
> > -                            [self.managedObjectContext save:nil];
> > -                            if (success) {
> > -                                success();
> > -                            }
> > -                        }
> > -                        failure:failure];
> > +    NSManagedObjectID *blogID = [blog objectID];
> > +    [self.managedObjectContext performBlock:^{
>
> The way services should be used is that the place that creates the service
> instance is responsible for wrapping the method call in the performBlock.
> So I would go back to the original implementation and make sure all the
> places that call this code (that aren't obvious to be on the main thread)
> wrap it in the performBlock there. @jleandroperez
> <https://github.com/jleandroperez> do you agree?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wordpress-mobile/WordPress-iOS/pull/4182/files#r38086245>
> .
>